controller_manager/src/controller_manager.cpp

```diff
@@ -1508,10 +1508,27 @@ void ControllerManager::activate_controllers(
     {
       RCLCPP_ERROR(
         get_logger(),
-        "After activation, controller '%s' is in state '%s' (%d), expected '%s' (%d).",
+        "After activation, controller '%s' is in state '%s' (%d), expected '%s' (%d). Releasing "
+        "interfaces!",
         controller->get_node()->get_name(), new_state.label().c_str(), new_state.id(),
         hardware_interface::lifecycle_state_names::ACTIVE,
         lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE);
+      controller->release_interfaces();
+      {
```

Comment:
Is there a reason to open an extra scope here?

Reply:
@firesurfer There is no specific reason. Just to scope the part of the switching interfaces separately. 

Do you want me to remove the scope? I don't mind removing it.

Reply:
No I don't mind. Just looked like an artifact from like a removed if-condition.

Reply:
@firesurfer I've removed the scope. Please check and approve it.

Thank you